### PR TITLE
T701-019: Fix zip_with_params on incomplete type decls.

### PR DESCRIPTION
**Pull request by Roldak**

### CODE REVIEW

File: user_manual/changes/T701-019.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+type: bugfix
+title: Fix ``zip_with_params`` on incomplete type decls
+description: |
+    This fixes an issue with calling ``P_Zip_With_Params`` on an discriminant
+    constraint that constrains the incomplete view of a discriminated type
+    would yield an empty result.
+title: 02-07-2020
```
Comment (pmderodat) [inline user_manual/changes/T701-019.yaml:7]: Pasto (`title` should be `date`), and the date format is wrong (`2020-07-02`)

File: ada/language/ast.py [hunk 2]
```diff
@@ -9913,7 +9913,7 @@ def zip_with_params():
             ._.formal_part.abstract_formal_params,
 
             lambda c=T.DiscriminantConstraint:
-            c.subtype.cast(T.TypeDecl)._.discriminants_list,
```
Comment (Roldak) [inline ada/language/ast.py:9916]: Not sure what this cast was for, I hope it won't break anything (it doesn't in our testsuite). Maybe you have an idea @raph-amiard ?

File: user_manual/changes/T701-019.yaml (hunk 1 above)
Comment (Roldak) [reply user_manual/changes/T701-019.yaml]: Thanks! I pushed too soon (testsuite wasn't done running :grimacing: ) Fixed!

File: ada/language/ast.py (hunk 2 above)
Comment (raph-amiard) [reply ada/language/ast.py:9916]: 1. I'm pretty sure the cast was wrong.
2. If it doesn't break anything in the testsuites, then I guess we'll see :)

**Pull request merged**